src/chat_statistics/graph.py

```diff
@@ -0,0 +1,145 @@
+                # differs from (userB, userA).
+                conections[(reply_from, reply_to)] += 1
+                # Increase the interaction  of anyone
+                # who has replayed a message or  received a reply
+                interactions[reply_from] += 1
+                interactions[reply_to] += 1
+
+        logger.info("Generating graph...")
+        # Set a value for our nodes.
+        # Here we consider the interaction of each
+        # person as the value of that node.
+        node_value = []
+        for user in users.keys():
+            # Consider the value 1 for Nodes that
+            # have not interacted with others.
+            if user not in interactions:
+                node_value.append(1)
+            # Consider the value of interaction + 1 for other nodes.
+            else:
+                node_value.append(interactions[user]+1)
+
+        sorted_node_value = sorted(node_value, reverse=True)
+        # Create color based on the amount of nodes.
+        colors = self.red2blue(len(users))
+        # Match the colors to the values of each node in this section.
+        value_color_dict = {}
+        for i in range(len(colors)):
+            value_color_dict[sorted_node_value[i]] = colors[i]
+        users_color = list(map(lambda x: value_color_dict[x], node_value))
+        # Generate graph
+        G = Network(height='100%', width='100%')
+        # Add all nodes to the graph.
+        G.add_nodes(list(users.keys()),
+                    title=list(users.values()),
+                    value=node_value,
+                    label=list(users.values()),
+                    color=users_color
+                    )
+        # Add all edges to the graph.
+        for conection in conections:
+            a, b = conection
+            nodes = G.get_nodes()
+            if a not in nodes or b not in nodes:
+                continue
+            G.add_edge(a, b)
+
+        # Generate the graph with the necessary options.
+        G.set_options('''
+                        var options = {
+                                        "nodes": {
+                                            "font": {
+                                            "size": 15
+                                            }
+
+                                        },
+                                        "physics": {
+                                            "barnesHut": {
+                                            "gravitationalConstant": -30000,
+                                            "centralGravity": 1.1,
+                                            "springLength": 500
+                                            },
+                                            "minVelocity": 0.75
+                                        }
+                                        }
+        ''')
+        # G.show_buttons(filter_=['physics','nodes'])
+        G.show(str(Path(output_dir).relative_to(Path.cwd()) / "graph.html"))
```

Comment:
Because this method takes relative path as inputs rather than absolute paths. 
To solve this problem, I also made `DATA_DIR` relative in the `src/data/__init.py' :
"""
DATA_DIR_ABSOLUTE = Path(__file__).resolve().parent
DATA_DIR = DATA_DIR_ABSOLUTE.relative_to(DATA_DIR_ABSOLUTE.parents[1])
""" 

So now we can use `DATA_DIR`, which is given as a input of method to the variable `output_dir` :
"""
G.show(str(Path(output_dir) / "graph.html"))
""" 
It should be noted that the `stat.py` also works well with the relative `DATA_DIR`.

Reply:
On my laptop, it works with `DATA_DIR_ABSOLUTE` as well. What error do you get if you run it with `DATA_DIR_ABSOLUTE` path?

Reply:
There should be no problem with absolute dir.

Reply:
@siniorone

Reply:
> On my laptop, it works with `DATA_DIR_ABSOLUTE` as well. What error do you get if you run it with `DATA_DIR_ABSOLUTE` path?
```
2021-12-15 21:56:52.359 | INFO     | __main__:generate_graph:76 - Generating graph...
Start : This command cannot be run due to the error: The system cannot find the file specified.
At line:1 char:1
+ Start "/home/siniorone/Github/telegram_statistics/src/data/graph.html ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Start-Process], InvalidOperationException
    + FullyQualifiedErrorId : InvalidOperationException,Microsoft.PowerShell.Commands.StartProces 
   sCommand
   ```

Reply:
I think this error is from PowerShell to open the HTML file. But the graph file is created correctly and stored in the desired location when the absolute path is used.

Reply:
OK. So let's stick to the absolute path (`DATA_DIR`) and get rid of the relative one. Using relative pathes is tricky and not a best practice.